### PR TITLE
HVAC mode in Modbus platform

### PR DESCRIPTION
**Pull request by avishorp**
## Proposed change

Add support for HVAC mode registers in Modbus integration

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/73906
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/modbus.markdown [hunk 1]
```diff
@@ -79,7 +79,7 @@ timeout:
   default: 5
   type: integer
 type:
-  description: Type of communication. Possible values are `tcp` Modbus messages with Modbus TCP frame on TCP/IP, `udp` Modbus messages with Modbus TCP frame on UDP, `rtuovertcp` Modbus messages with a wrapper TCP/IP simulating a serial line, 'serial' Modbus serial (RS485).
+  description: Type of communication. Possible values are `tcp` Modbus messages with Modbus TCP frame on TCP/IP, `udp` Modbus messages with Modbus TCP frame on UDP, `rtuovertcp` Modbus messages with a wrapper TCP/IP simulating a serial line.
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]: This change is not correct.

File: source/_integrations/modbus.markdown [hunk 2]
```diff
@@ -454,6 +466,54 @@ climates:
...
+      required: false
+      type: [map]
+      keys:
+        address: 
+          description: The address of the HVAC mode register.
+          required: true
+          type: integer
+        values:
+          description: A mapping between the register values and HVAC modes
+          required: true
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:480]: Why is this required, when all the entries above are optional ?

File: source/_integrations/modbus.markdown [hunk 3]
```diff
@@ -454,6 +466,54 @@ climates:
...
+          type: [map]
+          keys:
+            "off":
+              description: The register value corresponding to HVAC Off mode.
+              required: false
+              type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:486]: No default value ?

File: source/_integrations/modbus.markdown [hunk 4]
```diff
@@ -454,6 +466,54 @@ climates:
...
+            heat:
+              description: The register value corresponding to HVAC Heat mode.
+              required: false
+              type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:490]: No default value ?

File: source/_integrations/modbus.markdown [hunk 5]
```diff
@@ -454,6 +466,54 @@ climates:
...
+            cool:
+              description: The register value corresponding to HVAC Cool mode.
+              required: false
+              type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:494]: No default value ?

File: source/_integrations/modbus.markdown [hunk 6]
```diff
@@ -454,6 +466,54 @@ climates:
...
+            auto:
+              description: The register value corresponding to HVAC Auto mode.
+              required: false
+              type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:498]: No default value ?

File: source/_integrations/modbus.markdown [hunk 7]
```diff
@@ -454,6 +466,54 @@ climates:
...
+            dry:
+              description: The register value corresponding to HVAC Dry mode.
+              required: false
+              type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:502]: No default value ?

File: source/_integrations/modbus.markdown [hunk 8]
```diff
@@ -454,6 +466,54 @@ climates:
...
+            fan_only:
+              description: The register value corresponding to HVAC Fan only mode.
+              required: false
+              type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:506]: No default value ?

File: source/_integrations/modbus.markdown [hunk 9]
```diff
@@ -454,6 +466,54 @@ climates:
...
+            heat_cool:
+              description: The register value corresponding to HVAC Heat/Cool mode.
+              required: false
+              type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:510]: No default value ?

File: source/_integrations/modbus.markdown [hunk 10]
```diff
@@ -454,6 +466,54 @@ climates:
...
+    hvac_onoff_register:
+      description: Definition of a register holding and controlling the On/Off state of the climate device.
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]: Is "Definition" not "Address" ?

File: source/_integrations/modbus.markdown [hunk 11]
```diff
@@ -454,6 +466,54 @@ climates:
...
+        A zero value in this register means that the device is in off state, otherwise it is on. When writing
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]: It would be more logical to use True/False.

File: source/_integrations/modbus.markdown [hunk 12]
```diff
@@ -454,6 +466,54 @@ climates:
...
+        to this register, the value 0 will be written to turn the device off, 1 to turn it on.
+      required: false
+      type: integer
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:518]: should be a boolean

File: source/_integrations/modbus.markdown [hunk 13]
```diff
@@ -466,6 +526,12 @@ climates:
 | ------- | ----------- |
 | set_temperature | Set Temperature. Requires `value` to be passed in, which is the desired target temperature. `value` should be in the same type as `data_type` |
 
+### Service `modbus.set-hvac-mode`
+
+| Service | Description |
+| ------- | ----------- |
+| set_hvac_mode | Set HVAC mode. Requires `value` to be passed in, which is the desired mode. `value` should be a valid HVAC mode. A mapping between the desired state and the value to be written to the HVAC mode register must exist. Calling this service will also set the On/Off register to an appropriate value, if such a register is defined. |
+
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:536]: How does a user set-hvac-mode to off ?

**Review by janiversen — CHANGES_REQUESTED** [review]
I think there are code changes as well, will look at your updated PR later.

File: source/_integrations/modbus.markdown (hunk 1 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown]: This is a mistake, I shouldn't have touched this paragraph. I will revert it.

File: source/_integrations/modbus.markdown (hunk 2 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown:480]: If an HVAC mode register is defined, at least one value is optional here, but each individual one is optional.

File: source/_integrations/modbus.markdown (hunk 3 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown:486]: No. First, there is no standardization here between different HVAC manufacturers. Second, only those entries whose value is defined are considered valid modes, so, for example, if there are only entries for COOL and OFF, these will be the only possible modes shown to the user.

File: source/_integrations/modbus.markdown (hunk 4 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown:490]: See above.

File: source/_integrations/modbus.markdown (hunk 5 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown:494]: See above.

File: source/_integrations/modbus.markdown (hunk 9 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown:510]: See above.

File: source/_integrations/modbus.markdown (hunk 10 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown]: Yes, that is correct, I'll fix it.

File: source/_integrations/modbus.markdown (hunk 11 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown]: I don't understand what you mean. The address of the register is an integer. The value being read or written is also integer.

File: source/_integrations/modbus.markdown (hunk 12 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown:518]: It's an address.

File: source/_integrations/modbus.markdown (hunk 11 above)
Comment (janiversen) [reply source/_integrations/modbus.markdown]: Yes but the value can only be 0 or non-zero, which is the same as 0 == False and otherwise it is True. Using an integer makes you believe that non-zero values have different meanings.

File: source/_integrations/modbus.markdown (hunk 13 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown:536]: It's a mode like any other, and it works.

File: source/_integrations/modbus.markdown (hunk 11 above)
Comment (avishorp) [reply source/_integrations/modbus.markdown]: The configurable value is only an address. I've changed the description, hope it's not misleading now.

**Review by janiversen — DISMISSED** [review]
LGTM.

File: source/_integrations/modbus.markdown [hunk 14]
```diff
@@ -466,6 +528,12 @@ climates:
 | ------- | ----------- |
 | set_temperature | Set Temperature. Requires `value` to be passed in, which is the desired target temperature. `value` should be in the same type as `data_type` |
 
+### Service `modbus.set-hvac-mode`
```
Comment (frenck) [inline source/_integrations/modbus.markdown]:
```suggestion
### Service `modbus.set_hvac_mode`
```

**Review by frenck — APPROVED** [review]
Thanks, @avishorp 👍
